libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -1762,6 +1762,9 @@ const char *AP_Mission::Mission_Command::type() const {
     case MAV_CMD_DO_ENGINE_CONTROL:
         return "EngineControl";
     default:
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+        abort();
```

Comment:
Wouldn't panic be better here?

Reply:
If we changed this: https://github.com/ardupilot/ardupilot/blob/use-mav-result/libraries/AP_HAL_SITL/system.cpp#L34 to be abort() rather than an endless loop, yes, it would be strictly better IMO :-)

Having an `abort()` in there means that gdb pops up as soon as the error occurs.

Would there be general agreement on changing the infinite loop?

Reply:
@peterbarker how do you generate so many links that don't work?

My concern on the gdb pop up is that's not true unless it was in a debug instance already? Although to be fair if we weren't in a debug instance it's not like spinning really helped us....

Reply:
@WickedShell I have a computer generate those bad links for me :-)

Note it's asking for my branch in ardupilot's repo.  That's not right...

And yes, it only helps if you're running under GDB.  At the moment, at least...